source/common/grpc/google_grpc_utils.cc

```diff
@@ -133,8 +133,10 @@ GoogleGrpcUtils::channelArgsFromConfig(const envoy::config::core::v3::GrpcServic
 }
 
 std::shared_ptr<grpc::Channel>
-GoogleGrpcUtils::createChannel(const envoy::config::core::v3::GrpcService& config, Api::Api& api) {
-  std::shared_ptr<grpc::ChannelCredentials> creds = getGoogleGrpcChannelCredentials(config, api);
+GoogleGrpcUtils::createChannel(const envoy::config::core::v3::GrpcService& config, Api::Api& api,
```

Comment:
I think you might get push back here.  If a customer is using the grpc channel for xDS and *not* using AWS auth, they still need to pass in a cluster manager. 

Is this the only way to get the cluster manager into the signer?

Reply:
Yes, they still need to pass in a cluster manager even if they are not using AWS auth :( 

In `source/common/grpc/async_client_manager_impl.cc` L123 func `AsyncClientManagerImpl::factoryForGrpcService`, the switch case `kEnvoyGrpc (AsyncClientFactoryImpl)` was taking in the cluster manager `cm_` but not `kGoogleGrpc (GoogleAsyncClientFactoryImpl)`. With this change if I need the cluster manager I am forced to send the cluster manager all the way from `GoogleAsyncClientFactoryImpl` into AWS CredentialProviders.

Reply:
as of now I haven't discovered an alternate.